src/main/java/fi/aalto/cs/apluscourses/intellij/model/ScalaSdk.java

```diff
@@ -0,0 +1,87 @@
+package fi.aalto.cs.apluscourses.intellij.model;
+
+import com.intellij.openapi.application.WriteAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.roots.OrderRootType;
+import com.intellij.openapi.roots.libraries.LibraryTablesRegistrar;
+import com.intellij.openapi.util.io.FileUtilRt;
+import fi.aalto.cs.apluscourses.model.ComponentLoadException;
+import fi.aalto.cs.apluscourses.model.Library;
+import fi.aalto.cs.apluscourses.utils.DirAwareZipFile;
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Paths;
+import java.util.Objects;
+import org.apache.commons.io.FileUtils;
+import org.jetbrains.annotations.NotNull;
+
+public class ScalaSdk extends Library {
+
+  private static final String URL = "https://scala-lang.org/files/archive/";
+  private static final String ROOTS = "scala-compiler.jar scala-library.jar scala-reflect.jar";
+  @NotNull
+  private final Project project;
+
+  public ScalaSdk(@NotNull String name, @NotNull Project project) {
+    super(name);
+    this.project = project;
+  }
+
+  @Override
+  public void fetch() throws IOException {
+    File tempZipFile = createTempFile();
+    fetchZipTo(tempZipFile);
+    extractZip(tempZipFile);
+  }
+
+  @NotNull
+  private File createTempFile() throws IOException {
+    return FileUtilRt.createTempFile(getFileName(), ".zip");
+  }
+
+  private void fetchZipTo(File file) throws IOException {
+    FileUtils.copyURLToFile(new URL(URL + getFileName() + ".zip"), file);
+  }
+
+  private void extractZip(File file) throws IOException {
+    new DirAwareZipFile(file).extractDir(getFileName() + "/lib", getBasePath());
+  }
+
+  private String getFileName() {
+    return getName().replace("scala-sdk", "scala");
```

Comment:
why?

Reply:
`@NotNull`?

Reply:
"Why?" Scala plugin names Scala SDK libraries `scala-sdk-*`. When, in this approach, ZIPs are downloaded from scala-lang.org they are structured like
```
+/
+-+scala-12.3.1
  +-+lib
  | +--scala-compiler.jar
  | +--scala-library.jar
  | +--scala-reflect.jar
  +-+other
  +-+dirs
```
so there is the difference

Reply:
sorry, was not clear enough
> Scala plugin names Scala SDK libraries `scala-sdk-*` 

didn't know that there is this part in lib names